src/llfs/packed_array.hpp

```diff
@@ -116,6 +121,22 @@ struct PackedArray {
       out << "} PackedArray;" << std::endl;
     };
   }
+
+  void initialize_size_in_bytes(usize size_in_bytes)
+  {
+    this->size_in_bytes = size_in_bytes;
+    flags = static_cast<Flags>(flags | kSizeInBytesSet);
+  }
+
+  bool is_valid_size_in_bytes() const
+  {
+    return (this->flags & kSizeInBytesSet);
+  }
+
+  usize get_size_in_bytes() const
```

Comment:
Consider returning `Optional<usize>` to prevent coding errors involving forgetting to check whether the flag is set.

Reply:
will do the change.